src/test/java/org/prgms/locomocoserver/chat/application/MongoChatServiceTest.java

```diff
@@ -88,15 +102,26 @@ void saveChatMessage() {
         Long roomId = chatRoom.getId();
         Long senderId = creator.getId();
 
+        byte[] byteCode = imageToByteArray("src/test/java/org/prgms/locomocoserver/chat/application/스누피4.jpeg");
+        String imageBase64 = Base64.getEncoder().encodeToString(byteCode);
+        ChatMessageRequestDto requestDto = new ChatMessageRequestDto(roomId, senderId, "message", List.of(imageBase64));
+
         // when
-        mongoChatMessageService.saveChatMessage(roomId, new ChatMessageRequestDto(roomId, senderId, "message"));
-        boolean collectionExists = mongoTemplate.collectionExists("chat_messages_" + roomId);
+        List<String> imageUrls = chatImageService.create(requestDto);
+        mongoChatMessageService.saveChatMessageWithImage(roomId, imageUrls, requestDto);
+
         String collectionName = mongoChatMessageService.getChatRoomName(roomId);
+        boolean collectionExists = mongoTemplate.collectionExists(collectionName);
         long messageCount = mongoTemplate.getCollection(collectionName).countDocuments();
 
+        List<ChatMessageMongo> messages = mongoTemplate.findAll(ChatMessageMongo.class, collectionName);
+        assertThat(messages).isNotEmpty();
+        ChatMessageMongo lastMessage = messages.get(messages.size() - 1);
+
         // then
         assertThat(collectionExists).isTrue();
         assertThat(messageCount).isEqualTo(2);
```

Comment:
제 로컬에서 실행했더니 여기부분에서 예외가 나더라구요. 제가 실행할 때는 메시지가 1개였어서 그런 것 같아요. 따로 컬렉션 사이즈를 미리 테스트 코드 앞에서 계산한 뒤에 테스트 후 사이즈가 전보다 1 증가했는지 검증하는 것은 어떨까요? 예를 들면 아래처럼요?
```suggestion
        assertThat(messageCount).isEqualTo(beforeMessageSize + 1);
```

Reply:
좋은 생각인 것 같습니다 ! 수정하겠습니당